packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -264,82 +268,366 @@ export default class VrackMoveDialogCtrl {
+      ipv6List.forEach((ipv6) => {
+        this.loadIpRegion(ipv6.id);
+      });
+    }
+  }
+
+  updateOvhCloudConnect() {
+    if (this.data.eligibleServices?.ovhCloudConnect?.length > 0) {
+      this.data.eligibleServices.ovhCloudConnect = [];
+    }
+  }
+
+  getEligibleServices() {
+    return this.vrackService
+      .getEligibleServices(this.serviceName)
+      .then(({ data }) => {
+        if (!this.data.eligibleServices) {
+          this.data.eligibleServices = data.result;
+
+          SERVICES.forEach((service) => {
+            if (this.data.eligibleServices[service]) {
+              switch (service) {
+                case TYPE_SERVICE.cloudProject:
+                  if (this.data.eligibleServices?.cloudProject?.length > 0) {
+                    this.updateCloudProjectServiceInfo();
+                  }
+                  break;
+                case TYPE_SERVICE.dedicatedCloud:
+                  if (this.data.eligibleServices?.dedicatedCloud?.length > 0) {
+                    this.updateDedicatedCloudServiceInfo();
+                  }
+                  break;
+                case TYPE_SERVICE.dedicatedServer:
+                  if (this.data.eligibleServices?.dedicatedServer?.length > 0) {
+                    this.updateDedicatedServerServiceInfo();
+                  }
+                  break;
+                case TYPE_SERVICE.dedicatedServerInterface:
+                  if (
+                    this.data.eligibleServices?.dedicatedServerInterface
+                      ?.length > 0
+                  ) {
+                    this.updateDedicatedServerInterfaceServiceInfo();
+                  }
+                  break;
+                case TYPE_SERVICE.ip:
+                  if (this.data.eligibleServices?.ip?.length > 0) {
+                    this.updateIpServiceInfo();
+                  }
+                  break;
+                case TYPE_SERVICE.ipv6:
+                  if (this.data.eligibleServices?.ipv6?.length > 0) {
+                    this.updateIpv6ServiceInfo();
+                  }
+                  break;
+                case TYPE_SERVICE.legacyVrack:
+                  if (this.data.eligibleServices?.legacyVrack?.length > 0) {
+                    this.updateLegacyVrackServiceInfo();
+                  }
+                  break;
+                case TYPE_SERVICE.ipLoadbalancing:
+                  if (this.data.eligibleServices?.ipLoadbalancing?.length > 0) {
+                    this.updateIpLoadbalancingServiceInfo();
+                  }
+                  break;
+                default:
+                  this.updateOvhCloudConnect();
+                  break;
+              }
+            }
+          });
+        } else {
+          // add fields
+          this.updateEligibleServices(data.result);
+        }
+        if (data.status === API_STATUS.pending) {
+          this.$timeout(() => {
+            this.getEligibleServices();
+          }, ELIGIBLE_POLLING_INTERVAL);
```

Comment:
Why not use `$interval` instead ? What happens if you don't have any update for the status before the end of the timeout ?

Reply:
Because we made a polling on call API which send partially results and then a complete result.

Reply:
I think here should set the $timeout to destroy it when leaving the page 
this.poller = ...

Reply:
I'm not sure your solution is better, I think the $timeout is most useful here for what I'm doing, perhaps it's good idea for the task part but not here.

Reply:
Done